File: library/src/main/java/com/getstream/sdk/chat/model/AttachmentMetaData.java
package com.getstream.sdk.chat.model;

import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.getstream.sdk.chat.utils.Utils;

import java.io.File;
import java.util.Objects;

import io.getstream.chat.android.client.models.Attachment;

public class AttachmentMetaData {
    public boolean isSelected;
    public long videoLength;
    public long size;
    @Nullable
    public Uri uri;
    public String type;
    public String mimeType;
    public String title;
    @Nullable
    public File file;

    public AttachmentMetaData(Attachment attachment) {
        this.type = attachment.getType();
        this.mimeType = attachment.getMimeType();
        this.title = attachment.getTitle();
    }

    public AttachmentMetaData(@NonNull File file) {
        this.file = file;
        this.uri = Uri.fromFile(file);
        this.mimeType = Utils.getMimeType(file);
    }

    public AttachmentMetaData(@NonNull Uri uri, String mimeType) {
        this.uri = uri;
        this.mimeType = mimeType;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof AttachmentMetaData)) return false;
        AttachmentMetaData that = (AttachmentMetaData) o;
        return Objects.equals(uri, that.uri) &&
                Objects.equals(type, that.type) &&
                Objects.equals(mimeType, that.mimeType) &&
                Objects.equals(title, that.title) &&
                Objects.equals(file, that.file);
    }

    @Override
    public int hashCode() {
        return Objects.hash(uri, type, mimeType, title, file);
    }
}

File: library/src/main/java/com/getstream/sdk/chat/model/AttachmentMetaData.kt
package com.getstream.sdk.chat.model

import android.net.Uri
import com.getstream.sdk.chat.utils.Utils
import io.getstream.chat.android.client.models.Attachment
import java.io.File

data class AttachmentMetaData(
    var uri: Uri? = null,
    var type: String? = null,
    var mimeType: String? = null,
    var title: String? = null,
    var file: File? = null
) {
    var size: Long = 0
    var isSelected = false
    var videoLength: Long = 0

    constructor(attachment: Attachment) : this(
        type = attachment.type,
        mimeType = attachment.mimeType,
        title = attachment.title
    )

    constructor(file: File) : this(file = file, uri = Uri.fromFile(file)) {
        mimeType = Utils.getMimeType(file)
        type = getTypeFromMimeType(mimeType)
        size = file.length()
    }

    private fun getTypeFromMimeType(mimeType: String?): String = mimeType?.let { type ->
        when {
            type.contains("image") -> {
                ModelType.attach_image
            }
            type.contains("video") -> {
                ModelType.attach_video
            }
            else -> {
                ModelType.attach_file
            }
        }
    } ?: ModelType.attach_file
}
